float32 output
float32 time
string status